### PR TITLE
ConDec-375: Change tags in issue comments to JIRA like tags

### PR DESCRIPTION
**Pull request by Jcl15**
Switch from [tags] to {tags}
Still work in progress to see codacy issues

### CODE REVIEW

File: src/main/java/de/uhd/ifi/se/decision/management/jira/rest/KnowledgeRest.java [hunk 1]
```diff
@@ -45,6 +45,8 @@
 @Path("/decisions")
 public class KnowledgeRest {
 
+	public static boolean commentWriteLog;
```
Comment (lw2011) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/rest/KnowledgeRest.java]: did you mean to name the variable `commentWriteLock`? It is used as a lock.

File: src/test/java/de/uhd/ifi/se/decision/management/jira/extraction/model/TestComment.java [hunk 2]
```diff
@@ -364,14 +365,15 @@ public void testManuallyTaggingPro() {
...
 	@NonTransactional
 	public void testManuallyTaggingDecision() {
-		CommentImpl comment = getComment("[Decision]this is a manual pro tagged sentence [/Decision]");
+		CommentImpl comment = getComment("{Decision}this is a manual decision tagged sentence{Decision}");
+		System.out.println(comment.getTaggedBody(0));
 		// test the result in splits, fails when checked with equals
 		assertTrue(comment.getTaggedBody(0).contains("<span id=\"comment0\">"));
 		assertTrue(comment.getTaggedBody(0).contains("<span class=\"sentence Decision\"  id  = ui1>"));
 		// important that the tag is not inside the text area
 		assertTrue(comment.getTaggedBody(0).contains("<span class =tag>[Decision]</span>"));
```
Comment (lw2011) [inline src/test/java/de/uhd/ifi/se/decision/management/jira/extraction/model/TestComment.java]: Shouldn't `[Decision]` become `{Decision}`?

File: src/test/java/de/uhd/ifi/se/decision/management/jira/extraction/model/TestComment.java [hunk 3]
```diff
@@ -364,14 +365,15 @@ public void testManuallyTaggingPro() {
...
 		assertTrue(comment.getTaggedBody(0).contains("<span class = sentenceBody>"));
-		assertTrue(comment.getTaggedBody(0).contains("this is a manual pro tagged sentence "));
+		assertTrue(comment.getTaggedBody(0).contains("this is a manual decision tagged sentence"));
```
Comment (lw2011) [inline src/test/java/de/uhd/ifi/se/decision/management/jira/extraction/model/TestComment.java]: Shouldn't `[/Decision]` become `{Decision}`?

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/DecXtractEventListener.java [hunk 4]
```diff
@@ -90,7 +91,7 @@ private void handleDeleteIssue(DecisionKnowledgeElementImpl decisionKnowledgeEle
 	}
 
 	private void handleEditComment(DecisionKnowledgeElementImpl decisionKnowledgeElement) {
-		if(!KnowledgeRest.commentWriteLog) {
+		if(!DecXtractEventListener.editCommentLock) {//If locked, a rest service is manipulating the comment and should not be handled by EL
```
Comment (lw2011) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/DecXtractEventListener.java]: Is it really just "the" comment or could this also happen if in parallel another user updates some other comment?

Comment (Jcl15) [reply src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/DecXtractEventListener.java]: Good point. Need to think about that

File: src/main/resources/js/view.commentTab.toolbar-init.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
+/*
```
Comment (lw2011) [inline src/main/resources/js/view.commentTab.toolbar-init.js]: cool, I did not know we can use ES6 syntax. Could you post a link to atlassian's tutorial page you used for the code in this file?

File: src/main/resources/js/view.commentTab.toolbar-init.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                        entry.applyIfTextMode(addWikiMarkupCon).applyIfVisualMode(addRenderedContentCon);
+                    });
+
+                    entry.onUnregister(function cleanup() {
+                        conItem.remove();
+                    });
+                });
+            });
+
+function getHTMLLiItem(knowledgeType){
```
Comment (lw2011) [inline src/main/resources/js/view.commentTab.toolbar-init.js]: This and next functions blocks need indents.

File: src/main/resources/js/view.commentTab.toolbar-init.js (hunk 5 above)
Comment (Jcl15) [reply src/main/resources/js/view.commentTab.toolbar-init.js]: https://developer.atlassian.com/server/jira/platform/extending-the-rich-text-editor-in-jira/

**Pull request merged**